test/tools/router_check/router.cc

```diff
@@ -82,7 +82,12 @@ RouterCheckTool::RouterCheckTool(
     std::unique_ptr<Router::ConfigImpl> config, std::unique_ptr<Stats::IsolatedStoreImpl> stats,
     Api::ApiPtr api)
     : factory_context_(std::move(factory_context)), config_(std::move(config)),
-      stats_(std::move(stats)), api_(std::move(api)) {}
+      stats_(std::move(stats)), api_(std::move(api)) {
+  EXPECT_CALL(factory_context_->runtime_loader_.snapshot_,
+              featureEnabled(_, testing::An<const envoy::type::FractionalPercent&>(),
+                             testing::An<uint64_t>()))
+      .WillRepeatedly(testing::Invoke(this, &RouterCheckTool::runtimeMock));
```

Comment:
`Invoke` approach is necessary because of this statement [here](https://github.com/google/googletest/blob/master/googlemock/docs/ForDummies.md) `Important note: Google Mock requires expectations to be set before the mock functions are called, otherwise the behavior is undefined. In particular, you mustn't interleave EXPECT_CALL()s and calls to the mock functions.`

Reply:
I would make this an ON_CALL to setup a default action vs. an EXPECT_CALL